### ISSUE TITLE
change available languages at runtime

### ISSUE THREAD

**Issue by cany748** [nuxt-modules/i18n#2289]
### Describe the feature

there are times when i18n options need to be changed at runtime. now there is no way to do this, or I did not find how to do it. i see this one as Nuxt hook before initialization Vue I18n.

specifically, I need the functionality of limiting available languages based on information from `useRequestHeaders`

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution) (The contribution guideline of nuxt-modules/i18n is compliant with Nuxt too).
- [X] Check existing [discussions](https://github.com/nuxt-modules/i18n/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by kazupon** [comment]
Thank you for your feedback.
I backed from short vacation.

would you like to get more information from you about the use case with more detail and some codes.
we would not be able to understand your issue...

**Comment by cany748** [comment]
I want to have a nuxt hook named "i18n:config" that will allow me to extend and change the configuration of a module. just like on vite

**Comment by yazdanqm** [comment]
Hey guys ,
i have the same problem and i want set locales from my CMS with API request ..

**Comment by BobbieGoede** [comment]
Since the module relies on the build step to generate configs and routes it may not be possible to extend/add locales at runtime.. (at least right now). 

Limiting/disabling configured locales should be possible already using a middleware I think, could you provide some examples of what the usage of this feature would look like?

**Comment by yazdanqm** [comment]
> Since the module relies on the build step to generate configs and routes it may not be possible to extend/add locales at runtime.. (at least right now).
> 
> Limiting/disabling configured locales should be possible already using a middleware I think, could you provide some examples of what the usage of this feature would look like?

yes , we have a shop builder with some templates, we created some plugins for our users that one of them is multi language plugin. so users can turns the plugin on or off or they can select their default locale from the panel. i want to make default locale or i18n object dynamic. i think this is a useful feature

**Comment by gbyesiltas** [comment]
Yep, we have a very similar use case to @yazdanqm as well. This one would be nice to have, otherwise we need to do cross-checking in multiple places between our available locales and the complete set of locales defined in our nuxtI18n config

**Comment by Thy3634** [comment]
Lazy-load translations

nuxt-project/
├── lang/
│   ├── en-US.json
│   ├── [code].ts
├── nuxt.config.ts

**Comment by artem-schander** [comment]
> Lazy-load translations
> 
> nuxt-project/ 
> ├── lang/ │ 
>  │ ├── en-US.json
>  │ ├── [code].ts
> ├── nuxt.config.ts

does this work?

**Comment by randyhoulahan** [comment]
We have a head to a cms.   It is a united nations service.   same front head container for 197 sites, as it is a united nations service.   The default language changes on the back cms per site.   This is the only issue for use keeping head as one container instance for 197 sites.   

Will need to drop this lib or amend if possible to meet needs.  Is it possible?  have not jumped into your code base yet.

**Comment by maciej-baruch** [comment]
We are facing the same issue. 
ISR + CMS behind our content and right now the only way to update the languages is to rebuild the app.

**Comment by ikasianiuk** [comment]
> We are facing the same issue. ISR + CMS behind our content and right now the only way to update the languages is to rebuild the app.

same here :( 
managing locales & translations from CMS seems to be super common task. Any ETA for this feature maybe?

**Comment by randyhoulahan** [comment]
I rely on the cms for routing and prefix all locales now.  No default without prefix as a work around.
A compromise until a solution.

Thank You,

Randy J. Houlahan
Associate Information Systems Officer
Secretariat of the Convention on Biological Diversity  |  United Nations Environment Programme


From: Ira Kasianiuk ***@***.***>
Date: Monday, September 23, 2024 at 11:57 AM
To: nuxt-modules/i18n ***@***.***>
Cc: Randy Jacob Houlahan ***@***.***>, Comment ***@***.***>
Subject: Re: [nuxt-modules/i18n] change available languages at runtime (Issue #2289)

We are facing the same issue. ISR + CMS behind our content and right now the only way to update the languages is to rebuild the app.

same here :(
managing locales & translations from CMS seems to be super common task. Any ETA for this feature maybe?

**Comment by BobbieGoede** [comment]
I'm sure it is possible to implement this feature but due to limited time and few contributors other issues and (smaller) feature requests take priority. If this is a critical feature for your company I would be happy to review any pull request that works toward this functionality!

**Comment by randyhoulahan** [comment]
A PR was on my to-do list for before October if no one beat me to it,  but I am slightly behind now.

Maybe in the new year if no one beats me to it.

Thank You,

Randy J. Houlahan
Associate Information Systems Officer
Secretariat of the Convention on Biological Diversity  |  United Nations Environment Programme


From: Bobbie Goede ***@***.***>
Date: Monday, September 23, 2024 at 5:03 PM
To: nuxt-modules/i18n ***@***.***>
Cc: Randy Jacob Houlahan ***@***.***>, Comment ***@***.***>
Subject: Re: [nuxt-modules/i18n] change available languages at runtime (Issue #2289)

I'm sure it is possible to implement this feature but due to limited time and few contributors other issues and (smaller) feature requests take priority. If this is a critical feature for your company I would be happy to review any pull request that works toward this functionality!
